# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: accessible/xul/XULComboboxAccessible.cpp
# Commit: 006dad263530
# Full Hash: 006dad263530639c255256a61d13dd14b695d932
# Author: Morgan Rae Reschenberg <mreschenberg@mozilla.com>
# Date: 2025-04-15 04:07:32
# Description:
#   Bug 1959612: Verify focused doc can be retrieved before getting focused acc r=Jamie
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D245104
# ==============================================================================

diff -r b898e804320c -r 006dad263530 accessible/xul/XULComboboxAccessible.cpp
--- a/accessible/xul/XULComboboxAccessible.cpp	Mon Apr 14 17:00:47 2025 +0000
+++ b/accessible/xul/XULComboboxAccessible.cpp	Mon Apr 14 17:04:15 2025 +0000
@@ -172,7 +172,13 @@
 
     DocAccessibleParent* focusedDoc =
         DocAccessibleParent::GetFrom(focusedContext);
-    MOZ_ASSERT(focusedDoc && focusedDoc->IsDoc(), "No focused document found");
+    if (NS_WARN_IF(!focusedDoc)) {
+      // We can fail to get a document here if a user is
+      // performing a drag-and-drop selection with mouse. See
+      // `browser/base/content/tests/browser_selectpopup_large.js`
+      return LocalParent();
+    }
+    MOZ_ASSERT(focusedDoc->IsDoc(), "Got non-document?");
     focusedAcc = focusedDoc->AsDoc()->GetFocusedAcc();
   }
 
